tests/test_vsutil.py

```diff
@@ -110,3 +142,28 @@ def test_frame2clip(self):
         frame = self.WHITE_SAMPLE_CLIP.get_frame(0)
         clip = vsutil.frame2clip(frame)
         self.assert_same_frame(self.WHITE_SAMPLE_CLIP, clip)
+        # specifically test the path with disabled cache
+        vs.core.add_cache = False
+        black_frame = self.BLACK_SAMPLE_CLIP.get_frame(0)
+        black_clip = vsutil.frame2clip(black_frame, enforce_cache=True)
+        self.assert_same_frame(self.BLACK_SAMPLE_CLIP, black_clip)
+        # reset state of the core for further tests
+        vs.core.add_cache = True
+
+    def test_is_image(self):
+        """These are basically tests for the mime types, but I want the coverage. rooDerp"""
+        self.assertEqual(vsutil.is_image('something.png'), True)
+        self.assertEqual(vsutil.is_image('something.m2ts'), False)
+
+    def test_get_w(self):
+        self.assertEqual(vsutil.get_w(480, only_even=True), 854)
+        self.assertEqual(vsutil.get_w(480, only_even=False), 853)
+        self.assertEqual(vsutil.get_w(1080, 4 / 3), 1440)
+        self.assertEqual(vsutil.get_w(1080, 16 / 9), 1920)
+
+    def test_iterate(self):
+        def double_number(x: int) -> int:
```

Comment:
Oh and it should also check if the function is called the correct amount of times.

TLDR: Code coverage isn't everything

Reply:
But the result basically does that. If `2` is doubled `3` times, the result is `16`.